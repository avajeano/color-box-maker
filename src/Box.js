import React from "react";

const Box = ({ id, backgroundColor, width, height, removeBox }) => {
    // remove box compoent 
    const handleRemove = () => removeBox(id);

    return (
        <div
            className="Box"
            style={{
                backgroundColor : backgroundColor,
                width : `${width}px`,
                height : `${height}px`,
                position : "absolute"
            }}
        >
            <button onClick={handleRemove} style={{margin : '0% 0% 0% 100%' }}>X</button>
        </div>
    )
}

export default Box;