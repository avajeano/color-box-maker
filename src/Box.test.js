import React from "react";
import { render } from '@testing-library/react';
import Box from "./Box";

// Box smoke test
it('renders without crashing', () => {
    render(<Box id="test-id" backgroundColor="red" width={100} height={100} removeBox={() => {}} />)
})

// Box snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<Box id="test-id" backgroundColor="red" width={100} height={100} removeBox={() => {}} />);
    expect(asFragment()).toMatchSnapshot()
})