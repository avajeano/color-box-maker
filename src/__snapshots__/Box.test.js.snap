// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="background-color: red; width: 100px; height: 100px; position: absolute;"
  >
    <button
      style="margin: 0% 0% 0% 100%;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
